source/faas-automated-testing/faas-automated-testing.rst

```diff
@@ -0,0 +1,303 @@
+======================
+FaaS Automated Testing
+======================
+
+:Status: 
+:Minimum Server Version: 3.6
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+This specification is about the ability for drivers to automate tests for
+"Functions as a Service" from continuous integration.
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Specification
+=============
+
+Terms
+-----
+
+FaaS
+~~~~
+
+"Function as a Service", such as AWS Lambda.
+
+Implementing Automated FaaS Tests
+---------------------------------
+
+AWS Lambda
+~~~~~~~~~~
+
+This section describes the required setup of an AWS Lambda function and the
+steps needed to automate the deployment and execution of the function in
+Evergreen.
+
+Local Execution
+***************
+
+Prerequisites
+`````````````
+
+For the initial local setup the following are required:
+
+- The docker daemon running on the local machine.
+- The [https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html](AWS SAM CLI)
+
+The following environment variables MUST be present (See DRIVERS-2384 google doc for values):
+
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `AWS_REGION` - Set to us-east-1
+- `MONGODB_URI` - The local MongoDB instance
+
+Project Initialization
+``````````````````````
+
+Create the new project via SAM and follow the prompts:
```

Comment:
Does the binary support command line options that we can leverage to reduce the number of prompts? Looking at the prompts below, the only prompt that drivers will have to answer differently is that of the runtime, the rest could be hardcoded in the command.

Disregard this comment ff the command does not support options to skip some prompts.

Reply:
There is a `--no-interactive` option that can be used, but it does require creating a template config for every single language that needs to be supplied as a `--location` option. I wasn't thinking that would be part of this as I wanted each language to have control over the setup and not the spec. I'm happy though to do that if you think it's beneficial but will probably still require someone from each driver team to review those templates and make the necessary adjustments.